### PR TITLE
feat(vfg): convert field-select + more tests [khcp-11336]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

Convert VFG `FieldSelect` to use Kongponents + add more VFG component test coverage.
Addresses [KHCP-11336](https://konghq.atlassian.net/browse/KHCP-11336).

Adoption PR (with passing E2Es): https://github.com/Kong/konnect-ui-apps/pull/4209

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

[KHCP-11336]: https://konghq.atlassian.net/browse/KHCP-11336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/forms/src/components/fields/__tests__/FieldSelect.cy.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
+import type { FormSchema } from '../../../types'
+import FieldTester from '../../../../sandbox/FieldTester.vue'
+
+describe('<FieldTester /> - FieldSwitch', () => {
```
Comment (Leopoldthecoder) [inline packages/core/forms/src/components/fields/__tests__/FieldSelect.cy.ts:4]:
```suggestion
describe('<FieldTester /> - FieldSelect', () => {
```

**Comment by Leopoldthecoder** [comment]
Hi @kaiarrowood, for non-required select fields, previously we had a `No selection...` option so that users can unselect the field:

Kapture 2024-09-23 at 11 38 31

In this PR, the `No selection...` option is gone:

image


Could we either keep the `No selection...` option, or make such fields `clearable`?

Field schema for your reference:
```json
"algorithm": {
  "default": "round-robin",
  "description": "Which load balancing algorithm to use.",
  "one_of":
    [
      "round-robin",
      "lowest-latency",
      "lowest-usage",
      "consistent-hashing",
      "semantic"
    ],
  "type": "string"
}
```

**Comment by kaiarrowood** [comment]
> Hi @kaiarrowood, for non-required select fields, previously we had a `No selection...` option so that users can unselect the field:
> 
> Kapture 2024-09-23 at 11 38 31 [ Kapture 2024-09-23 at 11 38 31 ](https://private-user-images.githubusercontent.com/10095631/369745612-d17b263a-0974-418b-923a-19b6755aa75f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwOTc3NTEsIm5iZiI6MTcyNzA5NzQ1MSwicGF0aCI6Ii8xMDA5NTYzMS8zNjk3NDU2MTItZDE3YjI2M2EtMDk3NC00MThiLTkyM2EtMTliNjc1NWFhNzVmLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDEzMTczMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3NmZhMWZlMjM5ZDk2YjMzMTQxZTUxN2VkYjdhN2U3MjAyM2NkOWM5NjdjZmQ0OWNmNjNlNTIyMTk5MzI4NTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1dsxIT80W0qUKMhtL2gov27JtUKRMURFtcvwO11TH4s) [ ](https://private-user-images.githubusercontent.com/10095631/369745612-d17b263a-0974-418b-923a-19b6755aa75f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwOTc3NTEsIm5iZiI6MTcyNzA5NzQ1MSwicGF0aCI6Ii8xMDA5NTYzMS8zNjk3NDU2MTItZDE3YjI2M2EtMDk3NC00MThiLTkyM2EtMTliNjc1NWFhNzVmLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDEzMTczMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3NmZhMWZlMjM5ZDk2YjMzMTQxZTUxN2VkYjdhN2U3MjAyM2NkOWM5NjdjZmQ0OWNmNjNlNTIyMTk5MzI4NTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1dsxIT80W0qUKMhtL2gov27JtUKRMURFtcvwO11TH4s)
> 
> In this PR, the `No selection...` option is gone:
> 
> image
> Could we either keep the `No selection...` option, or make such fields `clearable`?
> 
> Field schema for your reference:
> 
> ```json
> "algorithm": {
>   "default": "round-robin",
>   "description": "Which load balancing algorithm to use.",
>   "one_of":
>     [
>       "round-robin",
>       "lowest-latency",
>       "lowest-usage",
>       "consistent-hashing",
>       "semantic"
>     ],
>   "type": "string"
> }
> ```

Thanks @Leopoldthecoder , I added this check for required on the `clearable` prop. Does this meet the requirements?

image

**Comment by Leopoldthecoder** [comment]
Yes, that looks good to me. I've also pulled the packages in KM and did some manual tests. Merging this PR.

**Review by Leopoldthecoder — APPROVED** [review]

**Pull request merged**